modules/apps/data-engine/data-engine-test/src/testFunctional/tests/macros/DataEngineUpgrade.macro

```diff
@@ -22,10 +22,79 @@ definition {
 	}
 
 	macro editAnUpgradedStructureWithFieldGroup {
+		fieldList = "Text,Date,Number";
 
-		// TODO LPS-126231 editAnUpgradedStructureWithFieldGroup pending implementation
+		WebContentNavigator.openWebContentAdmin(siteURLKey = "guest");
 
-		echo("empty stub");
+		NavItem.gotoStructures();
+
+		WebContentNavigator.gotoEditStructure(structureName = "My Structure with nested fields");
+
+		for (var structureFieldName : list "${fieldList}") {
+			DataEngine.editFieldLabel(
+				fieldFieldLabel = "${structureFieldName}",
+				fieldFieldLabelEdit = "${structureFieldName} Edited");
+		}
+
+		PortletEntry.changeLocale(locale = "pt-BR");
+
+		for (var structureFieldName : list "${fieldList}") {
+			DataEngine.editFieldLabel(
+				fieldFieldLabel = "${structureFieldName} - pt_BR",
+				fieldFieldLabelEdit = "${structureFieldName} - pt_BR Edited");
+		}
+
+		PortletEntry.changeLocale(locale = "ca_ES");
+
+		for (var structureFieldName : list "${fieldList}") {
+			DataEngine.editFieldLabel(
+				fieldFieldLabel = "${structureFieldName} - ca_ES",
+				fieldFieldLabelEdit = "${structureFieldName} - ca_ES Edited");
+		}
+
+		WebContentStructures.saveCP(structureName = "My Structure with nested fields");
```

Comment:
Hello @guivilarim , the setting of _structureName_ is only necessary when _addRoleViewPermissions_ is set. As you're creating the structures on the _guest_ group, you do not need to set this parameter. Thanks!